user/iplib/epoll.c

```diff
@@ -158,15 +158,19 @@ static struct event_queue *ep_get_alarm_evq(void)
  * some sort of user deferred destruction. (TODO). */
 static void ep_put_ceq_evq(struct event_queue *ceq_evq)
 {
+#if 0 /* TODO: EVQ/INDIR Cleanup */
 	ceq_cleanup(&ceq_evq->ev_mbox->ceq);
 	evq_remove_wakeup_ctlr(ceq_evq);
 	put_eventq_raw(ceq_evq);
+#endif
```

Comment:
On Wed, Oct 14, 2015 at 5:46 PM, Barret Rhoden <brho@cs.berkeley.edu> wrote:
> On 2015-10-14 at 16:00 Kevin Klues <notifications@github.com> wrote:
>> >     ceq_cleanup(&ceq_evq->ev_mbox->ceq);
>> >     evq_remove_wakeup_ctlr(ceq_evq);
>> >     put_eventq_raw(ceq_evq);
>> > +#endif
>>
>> Is the change here that the race triggered this code to break so you
>> just commented it out until you come up with a proper fix?
>
> Yep, that's the TODO in the ifdefs and above these two functions.  The
> fix is extremely complicated, and doing nothing just wastes memory.
>
> Barret
>
> p.s. - i had to manually add akaros@ to the CC here.

No you didn't.  I got the email twice because of that.  Github knows
to forward it back to the list if you respond.

>
> --
> You received this message because you are subscribed to the Google Groups "Akaros" group.
> To unsubscribe from this group and stop receiving emails from it, send an email to akaros+unsubscribe@googlegroups.com.
> To post to this group, send email to akaros@googlegroups.com.
> For more options, visit https://groups.google.com/d/optout.



-- 
~Kevin